server/libs/infra/src/repositories/partner.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { IPartnerRepository, PartnerIds } from '@app/domain';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PartnerEntity } from '../entities';
+
+@Injectable()
+export class PartnerRepository implements IPartnerRepository {
+  constructor(@InjectRepository(PartnerEntity) private readonly repository: Repository<PartnerEntity>) {}
+
+  getAll(userId: string): Promise<PartnerEntity[]> {
+    return this.repository.find({ where: [{ sharedWithId: userId }, { sharedById: userId }] });
+  }
+
+  get({ sharedWithId, sharedById }: PartnerIds): Promise<PartnerEntity | null> {
+    return this.repository.findOne({ where: { sharedById, sharedWithId } });
+  }
+
+  create({ sharedById, sharedWithId }: PartnerIds): Promise<PartnerEntity> {
+    return this.repository.save({
```

Comment:
Turns out this doesn't load relations for the users in the newly created entity.

We should change this to

```typescript
await this.repository.save(...);
return this.reposotory.findOrFail(...);
```